### PR TITLE
Expose host architecture for lit tests

### PR DESCRIPTION
**Pull request by Hardcode84**
Hack with manual LLVMConfig inclusion is already used in jit cmake so it shouldnt break more things.

### CODE REVIEW

File: tests/lit.site.cfg.in [hunk 1]
```diff
@@ -171,3 +172,5 @@ if lit.util.which('gdb', config.environment['PATH']):
         if LooseVersion(gdb_version) < LooseVersion('7.8'):
             gdb_dflags = '-dwarf-version=2'
     config.substitutions.append( ('%_gdb_dflags', gdb_dflags) )
+
+config.available_features.add(config.ldc_host_arch)
```
Comment (JohanEngelen) [inline tests/lit.site.cfg.in]: - please move this up such that it is grouped with the other available_features
- add a prefix `host_`.
- add a comment with examples of this feature (like we do for `target_*`).

**Comment by JohanEngelen** [comment]
Once merged, wiki page should be updated: https://wiki.dlang.org/LDC_Lit-based_testsuite#.22Features.22

**Comment by JohanEngelen** [comment]
Assuming this is only used sparingly, you may simply provide a simple fixup for LLVM <= 3.8 for which it currently doesn't work.

File: tests/lit.site.cfg.in (hunk 1 above)
Comment (Hardcode84) [reply tests/lit.site.cfg.in]: done

**Comment by JohanEngelen** [comment]
LGTM, assuming it actually works.

**Pull request merged**